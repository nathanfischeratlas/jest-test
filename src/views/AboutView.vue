<template>
  <div class="about">
    <h1>{{ description }}</h1>
  </div>
</template>

<script>
export default {
  name: 'AboutView',

  data() {
    return {
      description: ''
    }
  }
}
</script>
