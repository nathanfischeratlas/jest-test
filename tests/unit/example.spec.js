import { shallowMount, mount, createLocalVue } from '@vue/test-utils'
import router from '@/router'
import HelloWorld from '@/components/HelloWorld.vue'
import HomeView from '@/views/HomeView.vue'
import AboutView from '@/views/AboutView.vue'

const localVue = createLocalVue()

describe('All tests', () => {
  it('Renders props.msg when passed', () => {
    const msg = 'new message'
    const wrapper = shallowMount(HelloWorld, {
      propsData: { msg }
    })

    expect(wrapper.text()).toMatch(msg)
  })

  it('HelloWorld rendered correctly', () => {
    const wrapper = shallowMount(HelloWorld)

    expect(wrapper.exists()).toBe(true)
  })

  it('Snapshot HelloWorld', () => {
    const wrapper = mount(HelloWorld)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('HelloWorld renders with the correct class', () => {
    const wrapper = mount(HelloWorld)

    expect(wrapper.classes()).toContain('hello')
  })

  it('Click on first button', async () => {
    const wrapper = mount(HelloWorld)
    await wrapper.find('#button').trigger('click')

    expect(wrapper.vm.functionCalled).toBe(true)
  })

  it('Emits an custom event when second button is clicked', async () => {
    const wrapper = mount(HelloWorld)
    await wrapper.find('#buttonEvent').trigger('click')

    expect(wrapper.emitted().buttonClick).toBeTruthy()
  })

  it('Renders the correct button text', () => {
    const wrapper = mount(HelloWorld)

    expect(wrapper.find('#buttonEvent').text()).toBe('Custom Event')
  })

  it('Check if renders two buttons', () => {
    const wrapper = mount(HelloWorld)
    const buttons = wrapper.findAll('button')

    expect(buttons.length).toBe(2)
  })

  it('Route navigation occurs correctly', async () => {
    const wrapper = mount(HomeView, {
      localVue,
      router
    })
    await wrapper.vm.$router.push('/about')

    expect(wrapper.vm.$route.path).toBe('/about')
  })

  it('Description is changed correctly', async () => {
    const wrapper = mount(AboutView)
    await wrapper.setData({ description: 'This is an about page' })

    expect(wrapper.vm.description).toBe('This is an about page')
  })
})
