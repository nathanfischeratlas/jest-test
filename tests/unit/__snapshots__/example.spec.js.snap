// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All tests Snapshot HelloWorld 1`] = `
<div class="hello">
  <h1></h1>
  <p>
    For a guide and recipes on how to configure / customize this project,<br>
    check out the
    <a href="https://cli.vuejs.org" target="_blank" rel="noopener">vue-cli documentation</a>.
  </p>
  <h3>Installed CLI Plugins</h3> <button id="button">Click</button> <button id="buttonEvent">Custom Event</button>
</div>
`;
